Documentation/interact_v3.md

```diff
@@ -0,0 +1,182 @@
+$ etcdctl put foo1 bar1_new   # revision = 5
+```
+
+Here are an example to access the past versions of keys:
+
+```bash
+$ etcdctl get foo foo10 # access the most recent versions of keys
+foo
+bar_new
+foo1
+bar1_new
+
+$ etcdctl get --rev=4 foo foo10 # access the versions of keys at revision 4
+foo
+bar_new
+foo1
+bar1
+
+$ etcdctl get --rev=3 foo foo10 # access the versions of keys at revision 3
+foo
+bar
+foo1
+bar1
+
+$ etcdctl get --rev=2 foo foo10 # access the versions of keys at revision 2
+foo
+bar
+
+$ etcdctl get --rev=1 foo foo10 # access the versions of keys at revision 1
+```
+
+## Delete keys
+
+Applications can delete a key or a range of keys from etcd cluster.
+
+Here is the command to delete key `foo`:
+
+```bash
+$ etcdctl del foo
+1 # one key is deleted
+```
+
+Here is the command to delete keys ranging from `foo` to `foo10`:
+
+```bash
+$ etcdctl del foo foo10
+2 # two keys are deleted
+```
+
+## Watch changes of keys
+
+Applications can watch on a key or a range of keys to get any related modification updates.
+
+Here is the command to watch on key `foo`:
+
+```bash
+$ etcdctl watch foo 
+# in another terminal: etcdctl put foo bar
+foo
+bar
+```
+
+Here is the command to watch on a range key from `foo` to `foo10`:
+
+```bash
+$ etcdctl watch foo foo10
+# in another terminal: etcdctl put foo bar
+foo
+bar
+# in another terminal: etcdctl put foo1 bar1
+foo1
+bar1
+```
+
+## Watch historical changes of keys
+
+Applications may want to watch for historical changes of keys in etcd for various reasons. For example, it is not unusual that an application wants to receive all the modifications of a key. If the application always keep connection to etcd, then `watch` is good enough. However, application or the etcd may fail. If there is a change happens during the failure, the application will not receive the update in real time. To get the update, application needs to be able to watch the historical changes of keys. To do this, application can specify a historical revision just like reading past version of keys.
+
+Suppose we finished the following sequence of operations:
+
+``` bash
+etcdctl put foo bar         # revision = 2
+etcdctl put foo1 bar1       # revision = 3
+etcdctl put foo bar_new     # revision = 4 
+etcdctl put foo1 bar1_new   # revision = 5
+```
+
+Here is an example to watch the historical changes:
+```bash
+# watch for changes on key `foo` since revision 2
+$ etcdctl watch --rev=2 foo
+PUT
+foo
+bar
+PUT
+foo
+bar_new
+
+# watch for changes on key `foo` since revision 3
+$ etcdctl watch --rev=3 foo
+PUT
+foo
+bar_new
+```
+
+## Compact revisions
+
+As we mentioned, etcd keeps revisions so that applications can read past versions of keys. However, applications may want to compact revisions to avoid unbounded data growing. After a compaction, etcd removes historical revisions and release resources for future use. All revisions before the compacted revision will be unavailable for accessing.
```

Comment:
etcd keeps revisions so that applications can read past versions of keys. However, to avoid accumulating an unbounded amount of history, it is important to compact away past revisions. After a compaction, etcd removes historical revisions, releasing resources for future use. All superseded data with revisions before the compacted revision will be unavailable.